File: array/27.py
from typing import List
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) == 0: return 0
        i = 0
        idx = 0
        while i < len(nums):
            if nums[i] != val:
                nums[idx] = nums[i]
                idx += 1
            i += 1
        return idx

def main():
    a = [0,1,2,2,3,0,4,2]
    val = 2
    print(Solution().removeElement(a, val))
    for i in a:
        print(i)

if __name__ == "__main__":
    main()

File: leetcode-cpp/27.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int idx = 0;
        for (int i = 0; i < nums.size(); i++)
            if (nums[i] != val)
                nums[idx++] = nums[i];
        return idx;
    }
};

int main(){
    vector<int> nums({3,2,2,3});
    int val = 3;
    Solution a;
    cout << a.removeElement(nums, val) << endl;
    for (int i = 0; i <nums.size(); i++)
        cout << nums[i] << endl;
    return 0;
}
